crates/primitives/src/log.rs

```diff
@@ -89,7 +89,6 @@ pub struct Log<T = LogData> {
     /// The address which emitted this log.
     pub address: Address,
     /// The log data.
-    #[cfg_attr(feature = "serde", serde(flatten))]
     pub data: T,
```

Comment:
this will now produce incorrect results for ethereum json rpc . supporting standard ethereum json rpc is a primary goal of alloy

Reply:
that would prevent us from being generic over decoded log structs

Reply:
What is your suggested solution?

Reply:
for us, the current behavior is working as intended. we don't have an intended solution as serializing logs to any format other than JSON is not a goal of the library

Reply:
Will you accept a solution that preserves the current JSON serialization but fixes binary serde?

Reply:
yes, we would

Reply:
Done